File: Week 2 - Array, Strings, Command line arguements, functions/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <math.h>

void coleman_liau(string a);

int main(void)
{
    string s = get_string("Text: ");

    coleman_liau(s);
}

void coleman_liau(string a)
{
    int L_counter = 0;
    int W_counter = 1;
    int S_counter = 0;

    for (int i = 0, n = strlen(a); i < n; i++)
    {
        if ((a[i] >= 'a' || a[i] >= 'A') && (a[i] <= 'z' || a[i] <= 'Z'))
        {
            L_counter++;
        }

        else if (a[i] == ' ')
        {
            W_counter++;
        }

        else if (a[i] == '.' || a[i] == '!' || a[i] == '?')
        {
            S_counter++;
        }
    }

    float L = ((float) L_counter / (float) W_counter) * 100;
    float S = ((float) S_counter / (float) W_counter) * 100;

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }

    else
    {
        printf("Grade %i\n", index);
    }

    //test for word, sentences, and letter counters
    //printf("%i letter(s)\n%i word(s)\n%i sentence(s)\n", L_counter, W_counter, S_counter);

}

File: Week 6 - Python/readability.py
from cs50 import get_string


def main():
    sentence = get_sentence()

    # all = 0 except word counter to account for last word that does not have space after it
    word_counter, sent_counter, letter_counter = 1, 0, 0

    # looping over every char in sentence
    for char in sentence:
        if char.isalpha() == True:
            letter_counter += 1
        elif char == ' ':
            word_counter += 1
        elif char == '.' or char == '?' or char == '!':
            sent_counter += 1

    # average letters and sentences per 100 words
    L = letter_counter / word_counter * 100
    S = sent_counter / word_counter * 100

    # coleman-liau calc
    CL = round(0.0588 * L - 0.296 * S - 15.8)

    if CL < 1:
        print("Before Grade 1")
    elif CL >= 16:
        print("Grade 16+")
    else:
        print(f"Grade {CL}")


def get_sentence():
    a = get_string("Text: ")
    return a


main()